engine/src/main/java/org/terasology/engine/world/block/BlockComponent.java

```diff
@@ -54,4 +54,10 @@ public Vector3i getPosition(Vector3i dest) {
         dest.set(position);
         return dest;
     }
+
+    @Override
+    public void copy(BlockComponent other) {
+        this.position = new Vector3i(other.position);
+        this.block = other.block;
```

Comment:
Do we need to deep-copy `Block`?

Reply:
block is also a immutable object.

Reply:
https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/engine/world/block/Block.java has quite a few setters mutating internal state for an _immutable object_ ... 🤔